fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -405,165 +393,49 @@ public void commitTransaction(long transactionId, List<TabletCommitInfo> tabletC
-                                if (commitBackends != null && commitBackends.contains(tabletBackend)) {
-                                    // if the backend load success but the backend has some errors previously, then it is not a normal replica
-                                    // ignore it but not log it
-                                    // for example, a replica is in clone state
-                                    if (replica.getLastFailedVersion() < 0) {
-                                        ++successReplicaNum;
-                                    }
-                                } else {
-                                    errorBackendIdsForTablet.add(tabletBackend);
-                                    errorReplicaIds.add(replica.getId());
-                                    // not remove rollup task here, because the commit maybe failed
-                                    // remove rollup task when commit successfully
-                                }
-                            }
-
-                            if (successReplicaNum < quorumReplicaNum) {
-                                List<String> errorBackends = new ArrayList<String>();
-                                for (long backendId : errorBackendIdsForTablet) {
-                                    Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackend(backendId);
-                                    errorBackends.add(backend.getId() + ":" + backend.getHost());
-                                }
-
-                                LOG.warn("Fail to load files. tablet_id: {}, txn_id: {}, backends: {}",
-                                        tablet.getId(), transactionId,
-                                        Joiner.on(",").join(errorBackends));
-                                throw new TabletQuorumFailedException(tablet.getId(), transactionId, errorBackends);
-                            }
-                        }
-                    }
-                }
-            }
+            stateMachineList.add(stateMachine);
         }
 
         txnSpan.setAttribute("tables", tableListString.toString());
-        txnSpan.setAttribute("num_partition", numPartition);
+
         // before state transform
         TxnStateChangeCallback callback = transactionState.beforeStateTransform(TransactionStatus.COMMITTED);
         // transaction state transform
         boolean txnOperated = false;
 
         Span unprotectedCommitSpan = TraceManager.startSpan("unprotectedCommitTransaction", txnSpan);
+
         writeLock();
         try {
-            unprotectedCommitTransaction(transactionState, errorReplicaIds, tableToPartition,
-                    tableToInvalidDictCacheColumns, tableToValidDictCacheColumns,
-                    totalInvolvedBackends, db);
+            unprotectedCommitTransaction(transactionState, stateMachineList);
             txnOperated = true;
+            int numPartitions = 0;
```

Comment:
Why calculate of the numPartitions is put into write lock?

Reply:
updated